nn_surrogate_benchmark/bbob_sampler.py

```diff
@@ -53,17 +55,29 @@ def main():
             random_state=42,
         )
         rows = []
-        for x in tqdm(combinations):
-            f_value = function(x)
-            rows.append(
-                {
-                    **{f"x{i+1}": float(val) for i, val in enumerate(x)},
-                    "y": float(f_value),
-                }
-            )
+        if GET_GRADIENTS:
+            for x in tqdm(combinations):
+                f_value = function(x)
+                grad = approx_fprime(x, function, epsilon=1e-5)
```

Comment:
It would be nice to implement BBOB functions using torch e.g.
```
def rosenbrock_function(x: torch.Tensor, a: float = 1.0, b: float = 100.0) -> torch.Tensor:
    """
    2D Rosenbrock function.
    Global minimum at (1,1) with f(1,1)=0.
    """
    return (a - x[0])**2 + b * (x[1] - x[0]**2)**2
```
you can find already existing repo here: https://github.com/PatrikValkovic/BBOBtorch

Reply:
Nice idea! Although we are moving the BBOB sampler logic to the second repo, so the changes will be applied there